src/components/LoginPage.jsx

```diff
@@ -0,0 +1,153 @@
+import { Link, useNavigate } from "react-router-dom";
+import loginImg from "../assets/login-illu.svg";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import * as Yup from "yup";
+import { useDispatch } from "react-redux";
+import { login } from "../redux/user/userActions";
+import CryptoJS from "crypto-js";
+import { toast, Slide } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+const initialValues = {
+  email: "",
+  password: "",
+};
+
+const validationSchema = Yup.object().shape({
+  email: Yup.string()
+    .required("Email is required")
+    .email("Invalid email address"),
+  password: Yup.string()
+    .required("Password is required")
+    .min(8, "Password must be at least 8 characters long"),
+});
+
+// password decryption function
+const decryption = (password) => {
+  const decrypted = CryptoJS.AES.decrypt(
+    password,
+    import.meta.env.VITE_SECRET_KEY
+  ).toString(CryptoJS.enc.Utf8);
+  return decrypted;
+};
+
+// password encryption function
+const encryption = (password) => {
+  const encrypted = CryptoJS.AES.encrypt(
+    password,
+    import.meta.env.VITE_SECRET_KEY
+  ).toString();
+
+  return encrypted;
+};
+
+const LoginPage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { users } = JSON.parse(localStorage.getItem("userLoginData"));
+  let userLoginData;
+
+  const onSubmit = (values, { resetForm }) => {
+    const registeredUser = users.find((user) => user.email === values.email);
+    if (!registeredUser) {
+      toast.error("User does not exist", {
+        position: toast.POSITION.TOP_CENTER,
+        transition: Slide,
+        autoClose: 1000,
+      });
+      return;
```

Comment:
In `if-else` statement, when one of the conditions gets satisfied then no further conditions get evaluated. So `return` statement is not required.